crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -518,6 +527,24 @@ impl<'db> Specialization<'db> {
             ty.find_legacy_typevars(db, typevars);
         }
     }
+
+    pub(crate) fn ordering(self, db: &'db dyn Db, other: Self) -> Ordering {
+        self.generic_context(db)
+            .ordering(db, other.generic_context(db))
+            .then_with(|| {
+                self.types(db)
+                    .iter()
+                    .zip(other.types(db))
```

Comment:
Is it guaranteed that `self` and `other` have the same length?

Reply:
No, thanks for catching that.